contracts/programs/gumball-machine/src/state/mod.rs

```diff
@@ -99,8 +99,11 @@ pub struct GumballMachineHeader {
     pub extension_len: u64,
     pub max_mint_size: u64,
     pub remaining: u64,
+    // why is max_items a u64 when our indices are written as u32s...this could break, our index writing could overflow
+    // we should bound it to 32 or write indices as u64s
     pub max_items: u64,
```

Comment:
Wanted to raise this. The indices we write to the gumball machine account are size u32. But we allow max_items to have size bounded by u64. Since we write 0...`max_items` as indices on account, this could cause overflow. For now I set `smallest_uninitialized_index` to be a u64 to match. But we should probably be careful about this

Reply:
This is a good point. We can make this a u32, but need to byte align the struct